### PR TITLE
Rust: expose DataChannel string message as binary

### LINKED ISSUE #1269: Worker panic in Rust in WebRtcMessage::new
### Issue description

I'm seeing occasional panics in the call to [WebRtcMessage::new](https://github.com/versatica/mediasoup/blob/d7bee8a34cf0880de6508fc57be1c049375e88fc/rust/src/data_structures.rs#L1074) because of that unwrap. I never call that function myself, so I don't know where the invalid byte(s) would be coming from

[Here's](https://gist.github.com/PaulOlteanu/e87c49771df1c792caab8a97863d412c) a log from the panic.

### PR DESCRIPTION
**Pull request by ibc**
Fixes #1269

### Details

- Do not convert bytes to (UTF-8) string but let the user app do it based on `ppid` (`WebRtcMessage` type).

### CODE REVIEW

**Comment by ibc** [comment]
But then, the `data_producer.send()` and `data_consumer.send()` methods require an additional `ppid` as argument (same as in Node).

**Comment by ibc** [comment]
> But then, the `data_producer.send()` and `data_consumer.send()` methods require an additional `ppid` as argument (same as in Node).

Nope, not needed.

File: rust/tests/integration/direct_transport.rs [hunk 1]
```diff
@@ -444,13 +444,17 @@ fn send_with_subchannels_succeeds() {
 
         // Must be received by dataConsumer1 and dataConsumer2.
         direct_data_producer
-            .send(WebRtcMessage::String(both.to_string()), None, None)
+            .send(
+                WebRtcMessage::String(std::borrow::Cow::Borrowed(both.as_bytes())),
```
Comment (nazar-pc) [inline rust/tests/integration/direct_transport.rs]: What about importing `std::borrow::Cow` since it is used so many times here?

**Review by nazar-pc — APPROVED** [review]
Makes sense to me and this will definitely be a breaking change for `mediasoup` crate.

File: rust/tests/integration/direct_transport.rs (hunk 1 above)
Comment (ibc) [reply rust/tests/integration/direct_transport.rs]: It was already imported, my fault. I've simplified to `Cow::Borrowed` everywhere.

**Review by nazar-pc — APPROVED** [review]

**Comment by ibc** [comment]
> Makes sense to me and this will definitely be a breaking change for `mediasoup` crate.

https://github.com/versatica/mediasoup/pull/1290

**Pull request merged**